core/src/main/java/com/scalar/db/transaction/rpc/GrpcTransactionOnBidirectionalStream.java

```diff
@@ -64,12 +79,22 @@ public void onError(Throwable t) {
 
   @Override
   public void onCompleted() {
-    requestObserver.onCompleted();
+    requestStream.onCompleted();
   }
 
   private ResponseOrError sendRequest(TransactionRequest request) {
-    requestObserver.onNext(request);
-    return Uninterruptibles.takeUninterruptibly(queue);
+    requestStream.onNext(request);
+
+    ResponseOrError responseOrError =
+        Utility.pollUninterruptibly(
```

Comment:
Just a clarifying question.
The queue used here is empty or has one entry always?
If so, the reason for using the queue is to use `poll` to wait for data easily?